pyproject.toml

```diff
@@ -2,8 +2,8 @@
 # Minimum requirements for the build system to execute.
 # See https://github.com/scipy/scipy/pull/12940 for the AIX issue.
 requires = [
-    "meson-python==0.13.1",
-    "meson==1.2.1",
+    "meson-python>=0.13.1,<0.14",
+    "meson>=1.2.1,<2",
```

Comment:
This one looks right to me. Note that newer Meson versions have more shiny features one may want (e.g., you need 1.4.0 to have `dependency('numpy')` work with numpy 2.0 and up, so could be bumped a bit as well if desired.

Reply:
I vaguely remember trying to upgrade this in https://github.com/pandas-dev/pandas/pull/57895 and running into issues; we might want to keep the pin here below 1.4.0 since that has some fundamental changes to how NumPy is integrated (or on the flip side, bump the min to 1.4.0, although that would be a larger task)

Reply:
Feel free to bump and make that change as well :)